irohad/ametsuchi/impl/redis_flat_block_query.cpp

```diff
@@ -111,5 +127,54 @@ namespace iroha {
           });
     }
 
+    rxcpp::observable<model::Transaction> RedisFlatBlockQuery::getTxByHash(
+        std::string hash) {
+      return rxcpp::observable<>::create<model::Transaction>(
+          [this, hash](auto subscriber) {
+            auto blockId = this->getBlockId(hash);
+            if (not blockId) {
+              subscriber.on_completed();
+              return;
+            }
+
+            block_store_.get(blockId.value()) | [](auto bytes) {
+              return model::converters::stringToJson(bytesToString(bytes));
+            } | [this](const auto &json) {
+              return serializer_.deserialize(json);
+            } | [&](const auto &block) {
+              for (auto &tx : block.transactions) {
+                if (iroha::hash(tx).to_string() == hash) {
+                  subscriber.on_next(tx);
+                }
+              }
+            };
+            subscriber.on_completed();
+          });
+    }
+
+    boost::optional<model::Transaction> RedisFlatBlockQuery::getTxByHashSync(
+        std::string hash) {
+      auto blockId = this->getBlockId(hash);
+      if (not blockId) {
+        return boost::none;
+      }
+      return block_store_.get(blockId.value()) |
+          [](auto bytes) {
+            return model::converters::stringToJson(bytesToString(bytes));
+          }
+      | [this](const auto &json) { return serializer_.deserialize(json); } |
```

Comment:
is this proper formatting? D:

Reply:
My clang-format tells me everything is ok )